infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -508,6 +508,11 @@ public void updateSyncStatus(String clusterId, SyncType lastSyncType, SyncStatus
 
             CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> record = txn.getRecord(replicationStatusTable, key);
 
+            // the condition is true when FSM to a remote cluster has been shutdown.
+            if(record.getPayload() == null && status == SyncStatus.STOPPED) {
```

Comment:
the STOPPED status is also used for indicating other states as well: like remote leader loss, force snapshot sync. 
To ensure that the incoming STOPPED status corresponds to a shutdown FSM, we need to check the payload as well since we now delete the record in another path.